source/agora/node/Ledger.d

```diff
@@ -2131,7 +2156,8 @@ unittest
     assert(ledger.validateConsensusData(data) ==
         Ledger.InvalidConsensusDataReason.MayBeValid);
 
-    ledger.forceCreateBlock();
+    ledger.prepareNominatingSet(data, Block.TxsInTestBlock, mock_clock.networkTime());
+    ledger.externalize(data);
```

Comment:
Why this change?

Reply:
Now that we actually check the block `randomseed` in the unit test we need this so that the required `preimages` are revealed.

Reply:
But `forceCreateBlock` does both: https://github.com/bosagora/agora/blob/56b7560063cb9086cb925295f29bb6fc783d83e4/source/agora/node/Ledger.d#L1385-L1396

Reply:
We can not use `forceCreateBlock` here as we are not wanting all nodes to simulate revealing `preimages`.

Reply:
``` D
/// simulate block creation as if a nomination and externalize round completed
    private void forceCreateBlock (ulong max_txs = Block.TxsInTestBlock,
        string file = __FILE__, size_t line = __LINE__)
    {
        const next_block = this.getBlockHeight() + 1;
        simulatePreimages(this.enroll_man, next_block);
        ConsensusData data;
        this.prepareNominatingSet(data, max_txs, this.clock.networkTime());
        assert(data.tx_set.length >= max_txs);
        const expected_ts = this.params.GenesisTimestamp + data.time_offset;
        if (this.clock.networkTime() < expected_ts ||
            this.clock.networkTime() > (expected_ts + block_time_offset_tolerance.total!"seconds"))
        {
            if (auto mc = cast(MockClock) this.clock)
                mc.setTime(this.params.GenesisTimestamp + data.time_offset);
            else
                assert(0, "Need a MockClock or time handled correctly to call forceCreateBlock");
        }
        if (!this.externalize(data, file, line))
        {
            assert(0, format!"Failure in unit test. Block %s should have been externalized!"(
                       this.getBlockHeight() + 1));
        }
    }
```